File: src/api/MixServer.Domain/Queueing/Models/QueuePosition.cs
using MixServer.Domain.Queueing.Entities;

namespace MixServer.Domain.Queueing.Models;

public record QueuePosition(QueueItemEntity? Current, QueueItemEntity? Next, QueueItemEntity? Previous)
{
    public static QueuePosition Empty => new(null, null, null);
}

File: src/clients/mix-server-client/src/app/services/repositories/models/QueuePosition.ts
import {QueueItem} from "./queue-item";

export class QueuePosition {
  constructor(public current: QueueItem | null | undefined,
              public previous: QueueItem | null | undefined,
              public next: QueueItem | null | undefined) {
  }

  public static get Default(): QueuePosition {
    return new QueuePosition(null, null, null);
  }
}
